src/components/ViewRequest/Actions/ReworkRequest.tsx

```diff
@@ -7,28 +7,73 @@ import {
   DialogSurface,
   DialogTitle,
   DialogTrigger,
+  Field,
+  Textarea,
+  TextareaProps,
   Tooltip,
 } from "@fluentui/react-components";
 import { NavigateBackIcon } from "@fluentui/react-icons-mdl2";
-import { useRequest, useUpdateStage } from "api/requestsApi";
+import { useAddNote } from "api/notesApi";
+import {
+  UpdateRequestStage,
+  useRequest,
+  useUpdateStage,
+} from "api/requestsApi";
+import { useMyRoles } from "api/rolesApi";
 import { STAGES } from "consts/Stages";
+import { useState } from "react";
 import { useParams } from "react-router-dom";
 
+declare const _spPageContextInfo: {
+  userDisplayName: string;
+};
+
 const ReworkRequest = () => {
   const params = useParams();
   const requestId = Number(params.requestId);
   const request = useRequest(requestId);
   const updateStage = useUpdateStage();
+  const addNote = useAddNote(requestId);
+  const [reason, setReason] = useState("");
+  const myRoles = useMyRoles();
 
   const currentStage = STAGES.find(({ key }) => key === request.data?.stage);
 
+  let disableReworkButton = true;
+  if (currentStage?.next) {
+    if (currentStage?.key === "Draft") {
+      disableReworkButton = false;
+    } else if (myRoles.isHRL || myRoles.isCOSF) {
+      disableReworkButton = false;
+    } else if (
+      (request.data?.subStage === "OSFReview" ||
+        request.data?.subStage === "OSFPackageReview") &&
+      myRoles.isOSF
+    ) {
+      disableReworkButton = false;
```

Comment:
This substage doesn't exist anymore, should be "SelectionPackageOSFApproval" here and in the SendRequest component, so that OSF can send to Complete

Reply:
Maybe turn Stage and SubStage into enums -- or defined literals to keep it from being just aynthing -- and preventing possible issues where we changed and didn't catch. Might be others...